### PR TITLE
Feat: add issue templates

### PR DESCRIPTION
**Pull request by matthew-hagemann**
- Adds issue templates for bug reports (and when a bug should be reported to subiquity) and enhancement.
- Updates the README to distinguish between UI bugs and Backend bugs, which should be reported to subiquity.

You can preview what these options look like on my branch [here](https://github.com/matthew-hagemann/ubuntu-desktop-provision/issues/new/choose)

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 1]
```diff
@@ -0,0 +1,85 @@
+name: UI bug report for Ubuntu Desktop Provision
+description: Create a report to help us improve
```
Comment (rcaballeromx) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:2]: I'll make suggestions on some of the wording for clarity and conciseness. 
```suggestion
description: Report an issue to help the development
```

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+- type: markdown
+  attributes:
+    value: |
+      :note: This repository houses the Flutter UI used by the Ubuntu Installer. If your issue involves backend functionality, please report it under the Subiquity project here: [https://bugs.launchpad.net/ubuntu/+source/subiquity](https://bugs.launchpad.net/ubuntu/+source/subiquity).
+body:
+- type: checkboxes
+  attributes:
+    label: Is there an existing issue for this?
+    description: |
+      Please search to see if an issue already exists for this feature: https://github.com/canonical/ubuntu-desktop-provision/issues.
```
Comment (rcaballeromx) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:19]:
```suggestion
      Please check whether an issue already exists for this feature: https://github.com/canonical/ubuntu-desktop-provision/issues.
```

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    options:
+    - label: I have searched the existing issues
```
Comment (rcaballeromx) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:21]:
```suggestion
    - label: I have searched the existing issues and found none matching what I'm reporting.
```

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+      required: true
+- type: textarea
+  attributes:
+    label: Describe the bug
+    description: |
+      A clear and concise description of what the bug is. You can also attach screenshots or recordings.
```
Comment (rcaballeromx) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:27]:
```suggestion
      Describe the bug clearly and concisely. Include what is the bug and attach screenshots or recordings if needed.
```

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+  validations:
+    required: false
```
Comment (rcaballeromx) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:29]: I would make the description a required field.

File: .github/ISSUE_TEMPLATE/feature_request.yaml [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  - sminez
+  - matthew-hagemann
+body:
+- type: checkboxes
+  attributes:
+    label: Is there an existing issue for this?
+    description: |
+      Please search to see if an issue already exists for this feature: https://github.com/canonical/ubuntu-desktop-provision/issues?q=label%3Aenhancement.
+    options:
+    - label: I have searched the existing issues
```
Comment (rcaballeromx) [inline .github/ISSUE_TEMPLATE/feature_request.yaml:16]:
```suggestion
    - label: I have searched the existing issues and found none matching what I'm reporting.
```

File: .github/ISSUE_TEMPLATE/feature_request.yaml [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+- type: textarea
+  attributes:
+    label: Is the feature related to a problem or existing issue?
+    description: |
+      If so, please describe the problem or reference/link the issue number. If not, leave this empty.
+  validations:
+    required: false
+- type: textarea
+  attributes:
+    label: Describe the solution/feature you'd like.
```
Comment (rcaballeromx) [inline .github/ISSUE_TEMPLATE/feature_request.yaml:27]:
```suggestion
    label: Describe the feature, why the feature matters to you, and what benefits the feature will bring about.
```

**Review by sminez — APPROVED** [review]

**Pull request merged**